### PR TITLE
Refactor/use tanstack query

### PR DESCRIPTION
**Pull request by c0rydoras**
use tanstack/query for:
- data fetching
- data caching
- global error handling for queries (toast)

### CODE REVIEW

**Review by gianklug — COMMENTED** [review]
I feel like this shouldn't be here? Or is it automatically gone when building for prod?
image

**Comment by c0rydoras** [comment]
it is automatically gone when not in a dev env

File: src/index.tsx [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+    }),
+});
+
+render(
+    () => (
+        <QueryClientProvider client={queryClient}>
+            <App />
+            <Show when={import.meta.env.DEV}>
+                <SolidQueryDevtools initialIsOpen={false} />
+            </Show>
```
Comment (c0rydoras) [inline src/index.tsx:28]: @gianklug here we ensure the devtools dont end up in a prod build

Comment (gianklug) [reply src/index.tsx:28]: ack

**Comment by gianklug** [comment]
please add a "force refresh" button

**Comment by c0rydoras** [comment]
@gianklug where should i add the force refresh button? to every page? what should it forcefully refresh? the current kv. everything

**Comment by gianklug** [comment]
I'd suggest in the top right: 
image

I guess it should just refresh the current path in the context

**Comment by gianklug** [comment]
I feel like a refresh icon (in black) would fit the theme better

**Comment by c0rydoras** [comment]
I can do that but I would like to add icons in a separate PR

**Comment by gianklug** [comment]
Why is the icon animated only on load but not on click?

**Comment by c0rydoras** [comment]
@gianklug
> Why is the icon animated only on load but not on click?

this should now work as expected

**Comment by gianklug** [comment]
I trust in your words.

**Review by gianklug — APPROVED** [review]
LGTMwhatever

**Pull request merged**